chia/full_node/full_node.py

```diff
@@ -2152,9 +2152,18 @@ async def _replace_proof(
             if new_block is None:
                 continue
             async with self.db_wrapper.lock:
-                await self.block_store.add_full_block(new_block.header_hash, new_block, block_record)
-                await self.block_store.db_wrapper.commit_transaction()
-                replaced = True
+                try:
+                    await self.block_store.db_wrapper.begin_transaction()
+                    await self.block_store.add_full_block(new_block.header_hash, new_block, block_record)
+                    await self.block_store.db_wrapper.commit_transaction()
+                    replaced = True
+                except BaseException as e:
+                    await self.block_store.db_wrapper.rollback_transaction()
+                    log.error(
+                        f"_replace_proof error while adding block {block.header_hash} height {block.height},"
+                        f" rolling back: {traceback.format_exc()} {e}"
```

Comment:
If you intentionally want the 'message' at the end, ok.  I thought we often put it first.

```suggestion
                        f" rolling back: {e} {traceback.format_exc()}"
```

Reply:
So there is a related (but not quite right here) context manager now.

https://github.com/Chia-Network/chia-blockchain/blob/cbbdb6510a1f6f2dba096bb499897c124b8ffd94/chia/util/log_exceptions.py#L6-L11

I suggested to Chris that it should be renamed.  We could use that existing name for one that reraises the exception after logging.  Or have a parameter that defaults to `consume=False` or such along with a parameter for an optional custom message like you provide here.  Regardless, getting to standardized tooling used everywhere for this would help with consistency and maybe improve usage.